pallets/eth-bridge/src/eth.rs

```diff
@@ -237,8 +237,15 @@ fn process_send_response<T: Config>(result: Vec<u8>) -> Result<H256, DispatchErr
 }
 
 fn process_view_response<T: Config>(result: Vec<u8>) -> Result<i8, DispatchError> {
-    if result.is_empty() {
-        return Err(Error::<T>::InvalidDataLength.into())
+    let result_bytes = hex::decode(&result).map_err(|_| Error::<T>::InvalidViewResponseData)?;
+
+    if result_bytes.len() != 32 {
+        return Err(Error::<T>::InvalidViewResponseLength.into())
+    }
+
+    if !result_bytes[..31].iter().all(|&byte| byte == 0 || byte == 0xFF) {
+        return Err(Error::<T>::InvalidViewResponseValue.into())
```

Comment:
We expect the contract to return an i8 value between -1 and 1 and this function ensures the response conforms

Reply:
thanks @fluorostani , but how does it conform? Perhaps I'm missing something, but I think this checks if each byte is either 0 or 255. That means it will catch the whole number being `0` and `-1`, but I don't see how it catches `1` or, for that matter, how it would not accept things like `255`, `65535` or other more convoluted choices of particular bytes set at 255 while others are set at 0.

What am I missing?

Reply:
Sorry, to be more accurate, in the function in question we only sanity check Ethereum is sending the 32 byte form of an int8. It's then within check_tx_status that we specify the valid range:

```
pub fn check_tx_status<T: Config>(
   ...
        if let Ok(result) = make_view_call::<T>(calldata, &author) {
            match result {
                0 => return Ok(None),
                1 => return Ok(Some(true)),
                -1 => return Ok(Some(false)),
                _ => return Err(Error::<T>::InvalidEthereumCheckResponse.into()),
            }
   ...
   ```